### ISSUE TITLE
Tabs in diffs are not rendered correctly

### ISSUE THREAD

**Issue by vczh** [desktop/desktop#8616]
### Describe the bug

In both "Changes" and "History", when diffed lines has tab characters, the widths and positions are not correct.

### Version & OS

Version 2.2.3
Windows 10 1903 (18362.418)

### Steps to reproduce the behavior

1. Modify anything, be sure that the modified line should **begin with a tab character**
2. See diff in "Changes" (scenario 1)
3. Submit the change
4. See diff in "History" (scenario 2)

### Expected behavior

scenario1: The first tab character has 4 spaces in width
scenario2: The first tab character has 4 spaces in width

### Actual behavior

scenario1: The first tab character has 3 spaces in width
scenario2: The first tab character has 3 spaces in width
(if it is configured to be 8 spaces, then it it rendered in 7)

In this way, when the line has multiple tab characters, they all begin at (4n+3)-th character, instead of 4n.

I think it because the first "+" or "-" or " " takes 1 character space, which is not correct. These characters should not rendered like part of the text.

### Screenshots

N/A

### Logs

N/A

### Additional context

N/A

**Comment by billygriffin** [comment]
Hi @vczh, thanks for the issue. This seems to be an artifact of how Git produces unified diffs. I think the same problem also happens on the command line. Would you mind sharing how this is causing a problem for you and what you might able to do differently if there were 4 spaces after the + or -? It looks like copy/paste still works as expected from quickly testing that.

**Comment by vczh** [comment]
@billygriffin Hi, I use tabs to format my code a lot. By not rendering tabs in correct positions and widths, it is hard to review before submitting the code.

By the way, I don't think Git cause Github Desktop to have problems. You can preprocess Git's output before pasting all of the to the app.

One of the solution could be, render the first character and the rest separately, and do it line by line. When I select and copy something in diffs, the app don't actually give me those "+" and “-” characters. And when I just select the "+" character, and copy it, there is nothing in the clipboard. So it is not reason to put them in the same "text box" (**I also believe that be able to select "+" and "-" is a bug because of such behavior**), to let "+" and "-" affect how tabs are rendered.

**Comment by kuychaco** [comment]
@vczh thanks for the report. I'm trying to wrap my head around the issue you are seeing and could use some clarification.

I'm not seeing a discrepancy between my editor and GitHub Desktop:

<img width="540" alt="README_md_—_desktop-tutorial_and_GitHub_Desktop" src="https://user-images.githubusercontent.com/7910250/68809281-70d5c780-0620-11ea-93da-b4c54cc87f00.png">

Could you please provide screen shots of what you are observing? That would help us understand your issue.

**Comment by lee-dohm** [comment]
Steps to reproduce:

1. In a git repository associated with GitHub Desktop, create a new file using VSCode (v1.40.1)
1. _Before_ entering any text, ensure that the indentation settings for the file are `Tab Size: 4`
1. Enter the sample text:
    ```text
    sdf
    foo
    bar
    baz

    ABCD[TAB]1234
    ABC[TAB][TAB]1234
    ```
1. Save the file as `test.txt`
1. Open the repository in GitHub Desktop (v2.2.3)
1. View the diff

<img width="515" alt="Screen Shot 2019-11-13 at 3 11 18 PM" src="https://user-images.githubusercontent.com/1038121/68812677-851dc280-0628-11ea-9036-081290080946.png">

The difference in behavior is "tabs as characters" and "tab stops". In VSCode, a `Tab` character is interpreted as advancing to the next multiple of the tab width (tab stop behavior). In GitHub Desktop, a `Tab` character is interpreted as some number of space characters, no matter where the `Tab` character appears on the line (tabs as characters behavior).

**Comment by vczh** [comment]
@lee-dohm  is right
@kuychaco I don't know why you cannot repro the issue since I can do this consistently, I guess the reason could be your editor replaces tab to spaces in that file. This happens in VSCode too due to different configurations for different file types.

**Comment by lee-dohm** [comment]
Yes, I apologize, my understanding of the GitHub Desktop logic is incomplete. I made incorrect assumptions about why the code was behaving the way it was. Let's focus on properly describing the behavior please @vczh. The more exact information you can give us, the better we'll be able to understand what it is you're suggesting we should fix.

I created a new file with better content to see the alignment:

```text
sdf
foo
bar
baz

1234567890123456789012345678901234567890
ABCD[TAB]1234
ABC[TAB][TAB]1234
```

Which shows up like this:

<img width="562" alt="Screen Shot 2019-11-13 at 4 27 56 PM" src="https://user-images.githubusercontent.com/1038121/68816218-9ff53480-0632-11ea-90a0-bb4b6f946bf2.png">

**Comment by outofambit** [comment]
Thanks for the discussion everyone. I think the bug is best summarized as "the diff in desktop doesn't match the text as shown in a text editor". (As a side note, Desktop's behavior here is indeed in line with the git CLI.) Marking this as a bug.

**Comment by vczh** [comment]
What's the chance that a P3 bug get fixed?

**Comment by outofambit** [comment]
@vczh Not likely in the near future. This can change if we learn more about the impact of this bug, but right now it seems primarily cosmetic and has an easy workaround of checking the original file (hence P3). There's potential for this to become a `help-wanted` issue in the future, but we're in the middle of revising our process for that.

I'm leaving this issue opened and have labelled it a bug because it is something that causes our users pain/confusion and is something that we can fix. I know this probably isn't the answer you're looking for, but I appreciate you asking and I want to clearly set expectations.

**Comment by ahmetsait** [comment]
This issue seems to be fixed in #17428 and can be closed.

**Comment by tidy-dev** [comment]
Fixed in as mentioned in previous comment
